src/runners/rfqm_meta_transaction_consumer_runner.ts

```
@@ -0,0 +1,134 @@
+    SWAP_QUOTER_OPTS,
+} from '../config';
+import { METRICS_PATH, PROTOCOL_FEE_UTILS_POLLING_INTERVAL_IN_MS } from '../constants';
+import { getDBConnectionAsync } from '../db_connection';
+import { logger } from '../logger';
+import { RfqmService } from '../services/rfqm_service';
+import { providerUtils } from '../utils/provider_utils';
+import { RfqBlockchainUtils } from '../utils/rfq_blockchain_utils';
+import { SqsConsumer } from '../utils/sqs_consumer';
+
+import { getAxiosRequestConfig } from './http_rfqm_service_runner';
+
+const RFQM_JOB_DEQUEUED = new Counter({
+    name: 'rfqm_job_dequeued',
+    help: 'An Rfqm Job was pulled from the queue',
+});
+
+const RFQM_JOB_SUCCEEDED = new Counter({
+    name: 'rfqm_job_succeeded',
+    help: 'An Rfqm Job succeeded',
+});
+
+process.on('uncaughtException', (err) => {
+    logger.error(err);
+    process.exit(1);
+});
+
+process.on('unhandledRejection', (err) => {
+    if (err) {
+        logger.error(err);
+    }
+});
+
+if (require.main === module) {
+    (async () => {
+        // Start the Metrics Server
+        startMetricsServer();
+
+        // Build dependencies
+        const provider = providerUtils.createWeb3Provider(defaultHttpServiceWithRateLimiterConfig.ethereumRpcUrl);
+        const contractAddresses = await getContractAddressesForNetworkOrThrowAsync(provider, CHAIN_ID);
+        const quoteRequestor = new QuoteRequestor(
+            RFQT_MAKER_ASSET_OFFERINGS,
+            RFQM_MAKER_ASSET_OFFERINGS,
+            Axios.create(getAxiosRequestConfig()),
+            undefined, // No Alt RFQM offerings at the moment
+            logger.warn.bind(logger),
+            logger.info.bind(logger),
+            SWAP_QUOTER_OPTS.expiryBufferMs,
+        );
+        const protocolFeeUtils = ProtocolFeeUtils.getInstance(
+            PROTOCOL_FEE_UTILS_POLLING_INTERVAL_IN_MS,
+            ETH_GAS_STATION_API_URL,
+        );
+        const metaTxWorkerRegistry = META_TX_WORKER_REGISTRY || NULL_ADDRESS;
```

We should never use `NULL`, can we raise an error and exit here, if an address is not provided?

---

In this commit, we agreed to raise an error in the constructor of the `rfqmService` itself: https://github.com/0xProject/0x-api/pull/550/files#r621677486

Technically, an error will get raised, but I think it's better to move the error out of the constructor and into the bootstrap process. Will try doing that